### ISSUE TITLE
Unable to login

### ISSUE THREAD

**Issue by Kemmey** [Kemmey/Kemmey-TeslaWatch-Public#123]
Hi，

I follow your instructions to install both iphone and iwatch beta version testflight, and successfully communicate both devices and showed login in testflight, but the watchfortesla app still can not function(unable to login)!

pls help!

 

__Originally posted by @RogerHungYi in https://github.com/Kemmey/Kemmey-TeslaWatch-Public/issues/23#issuecomment-688176034__

**Comment by Kemmey** [comment]
@RogerHungYi, please post screenshots describing your issue and versions. 

Have you successfully installed both watch app for Tesla and watch app for Tesla sign-in assist from TestFlight?

Have you successfully logged in to sign-in assist on phone?

Have you installed the sign-in assist watch app?

Have you successfully transferred your token to your watch with sign-in assist?

**Comment by RogerHungYi** [comment]
The answers of all your questions are "yes".
But the watch for Tesla app still can not login.
![image](https://user-images.githubusercontent.com/70886708/92393793-06fc5080-f153-11ea-991d-ae273a8f2385.jpeg)![image](https://user-images.githubusercontent.com/70886708/92393817-0f548b80-f153-11ea-93f4-bfabbf5e2cad.jpeg)![image](https://user-images.githubusercontent.com/70886708/92393832-154a6c80-f153-11ea-9a13-f94bc2bbfa0a.jpeg)

**Comment by RogerHungYi** [comment]
Hi,
May you help me solve my problem?
Maybe I did something wrong,plz help.
Thanks a lot!

**Comment by Kemmey** [comment]
Hey, sorry for the long response time!

In the screenshots, your vehicle is offline - I believe for the watch app for tesla to accept the token, the vehicle needs to be online. That's actually a bug, as it should not be neccessary. 

Could you try the sign-in assist app again, but verify that your vehicle is online? Perhaps wake it up with your tesla iphone app. 

When the token is successfully transferred, you should be able to open the watch for tesla app - it will initially show the login screen - just leave it there for up to 30 seconds - it should start polling for data automatically. 

If that fails completely, please try this testflight: 

https://testflight.apple.com/join/m9aH0r8k￼

This app runs with the same token and the same engine as watch app for tesla, but has an ios counterpart - first verify that the ios counterpart can read data from your vehicle and properly show your vehicle location - then verify the same is true for the watch counterpart. 

Let me know how that goes. 

Oh - finally - please include screenshots of the watch app for tesla about screen, showing the build number - I have a lot of builds out and it's a bit hard to keep track of which build what error occurs :-D

**Comment by RogerHungYi** [comment]
Hi,
I followed your suggestions to get the vehicle online, then successfully transferred the token.
Then I turn on the watch for Tesla app and wait up till one minute.
But there's no responding.
And the new TestFlight which you gave me, it doesn't work.
Please help!
Thank you so much!
![image](https://user-images.githubusercontent.com/70886708/92852865-c71ebd00-f421-11ea-974d-3b80e464ede3.jpeg)
![image](https://user-images.githubusercontent.com/70886708/92852900-cf76f800-f421-11ea-8e73-36f2835ae5e0.jpeg)

**Comment by RogerHungYi** [comment]
Hi，

Would you pls help me solve my "unable to login" problem, though I purchased this app for almost one month long but still can't function!

Thanks

**Comment by Kemmey** [comment]
@RogerHungYi Hey - SO SORRY for the delayed response. 

From the screenshots of sign-in assist everything SHOULD be working, as the sign-in app can read your account just fine. That app runs the exact same code as watch app for Tesla, so it makes zero sense if one app can read and the other can't. 

It's almost as if the watch app for Tesla is running a really old version, that is not aware of the shared token logic. Could you try completely removing only the watch app for Tesla from your watch and reinstalling from the AppStore directly on the watch to ensure you're running the latest version?

**Comment by Kemmey** [comment]
And the other TestFlight like is somehow broken by github - I'll try again: https://testflight.apple.com/join/m9aH0r8k

**Comment by Kemmey** [comment]
Yes - that seemed to work - please try to join that TestFlight and verify if you can use that app on your phone. 

Again, I'm so sorry for your troubles with this - and I thank you for hanging in there. 

Please, if you want, request a refund from Apple and join the TestFlight beta until all this is properly resolved. 

https://testflight.apple.com/join/FdzFOLZv

**Comment by RogerHungYi** [comment]
Hi Kemmey,
Thanks for your prompt reply! 
The "where's tesla" app which you provide me to install, it works!!!!!!👍
But when I try to login the "watch for tesla" app again, it still can not login successfully! 
For now I can use where's tesla app without problem, but in the end I still wish to use watch for tesla app.  
I will not consider for refund as I still hope you can help me to solve this unable to login problem!
FYI, I already removed watch for tesla app from my apple watch for many times and reload It again, but same result.

thanks

**Comment by Kemmey** [comment]
Hey @RogerHungYi that's is so weird. Both apps use the exact same engine - literally, the network code is identical. 

Did you install the Watch app for Tesla from TestFlight as well?

https://testflight.apple.com/join/FdzFOLZv

**Comment by Kemmey** [comment]
@RogerHungYi Oh - and did the where's Tesla work on the watch as well?

**Comment by RogerHungYi** [comment]
Kemmey，

The where's tesla app works very well on my watch now!

**Comment by RogerHungYi** [comment]
@Kemmey but the watch for tesla app is still not function!

**Comment by RogerHungYi** [comment]
@Kemmey i did update watch for tesla app via Testflight!And try to login again from userID and password, but watch still shows error occurred!

**Comment by Kemmey** [comment]
@RogerHungYi Hmm - but if where's Tesla is logged in, watch for Tesla should pick up that token and use it. Try to open where's Tesla on watch, verify vehicle is online and has new data. Then open watch for Tesla, and leave it on the login screen for up to a minute - it should just switch to vehicle view automatically.

**Comment by RogerHungYi** [comment]
@Kemmey  I follow your instruction but  watch for tesla is still not working, and the where's Tesla working fine!

**Comment by Kemmey** [comment]
Hey @RogerHungYi I just now pushed an updated build in TestFlight. It will show the build version at the bottom of the login screen. Please ensure that it does so and the the build is 1.0.7 build 280. 

Now the login form should show more detailed error messages - please try to login with username and password and then scroll to top or bottom and send me a screenshot of the error message. Ensure you do NOT include your username in the screenshot for privacy reasons. 

I'm hoping with the added debug output we'll get closer to what the issue is here.

**Comment by RogerHungYi** [comment]
Hi Kemmey,
I can not see any new version of testflight app on my phone!
Pls see the photo attached!
![image](https://user-images.githubusercontent.com/70886708/94236038-1f37e200-ff3f-11ea-999e-fdf1a11680fb.jpeg)

The error message is still the same as previous!
Pls see the photos below!
![image](https://user-images.githubusercontent.com/70886708/94236142-4d1d2680-ff3f-11ea-8966-f99ff3e032ad.jpeg)
![image](https://user-images.githubusercontent.com/70886708/94236155-53130780-ff3f-11ea-83f5-721ef0dc23ad.jpeg)

**Comment by RogerHungYi** [comment]
Hi kemmey,
I updated the watchfortesla app in TestFlight as you mentioned and the version is different than what you indicted! 
However I still conduct sign-in assist for request token and then I login watchfortesla app via account number, finally it shows the message as below!
![image](https://user-images.githubusercontent.com/70886708/94253281-a3965f00-ff57-11ea-8f08-5bcd2a078cc2.jpeg)
![image](https://user-images.githubusercontent.com/70886708/94253294-ac873080-ff57-11ea-9f46-a0780f969de8.jpeg)
I hope the message can be helpful!

**Comment by Kemmey** [comment]
That is extremely helpful!

But wait - you said where's my tesla was working? Was that on the watch as well? 

Basically, the error indicates that either your watch in general, or just the tesla watch app, cannot access the internet at all.

**Comment by Kemmey** [comment]
can you try any of the tricks in this article to verify your watch has general internet access?

https://www.macrumors.com/how-to/browse-the-internet-watchos-5/

**Comment by RogerHungYi** [comment]
Yes, where's my tesla working very well both on my watch and iPhone!

Since my watch can run where's my tesla app very well, it mean no problem access internet! Am I right?

**Comment by RogerHungYi** [comment]
I will switch my watch from series3 to series 6 recently, I hope new watch can perform differently!

**Comment by Kemmey** [comment]
You are correct - if where's Tesla works on watch, your watch is properly connected. 

I absolutely do not know what is going on here - I have never heard of a single app being blocked from accessing internet resources while another works fine. 

It just does not make sense. On iOS it is possible to restrict an app from mobile data - but not on the watch, as far as I can tell. 

And both apps run the EXACT same network code. Literally copy and paste - same code.

**Comment by Kemmey** [comment]
@RogerHungYi Might be a long shot - but could you check this in settings - it IS possible to block a single app from network:

![image](https://user-images.githubusercontent.com/4688084/94359161-b9199f00-00a5-11eb-83f9-875b168c9fc2.jpeg)

![image](https://user-images.githubusercontent.com/4688084/94359162-bc148f80-00a5-11eb-8131-f831d24b9d20.jpeg)

![image](https://user-images.githubusercontent.com/4688084/94359165-be76e980-00a5-11eb-8285-3b07a192d646.jpeg)

**Comment by RogerHungYi** [comment]
![image](https://user-images.githubusercontent.com/70886708/94394649-6e288600-0190-11eb-817b-6cadb24d5f24.jpeg)

**Comment by Kemmey** [comment]
@RogerHungYi I just did an experiment based on your screenshot. 

I put my iPhone in powersave mode and turned off WiFi. Then turned off the screen. 

Now my watch cannot get online - at all. 

As soon as I unlock my phone, the watch comes back online. 

Could you try turning off powersave mode and try again?

**Comment by RogerHungYi** [comment]
Hi Kemmey,
In my iPhone setting, I can not find "Apps Only on Watch"! You may see the screen shot which I posted above! 

Therefore I can not find any watch for tesla icon in my iPhone.

**Comment by RogerHungYi** [comment]
Kemmey，
Before I turn off powersave mode, I delete watch app for tesla from my watch, I intend to install watch app for tesla again from app store, but strangely I found the app disappear from my purchased app and request another payment for this app!!! What's happened? Can you help?

**Comment by Kemmey** [comment]
@RogerHungYi That's just Apple being Apple... they literally have a knowledge base article detailing h this. They don't properly show already purchased apps on the watch App Store...

Please just tap purchase again - it will not bill you a second time but just download it again. If they DO bill you a second time, it would be because you've used different accounts in the App Store. In that case, just ask Apple for a refund - they always grant refunds, no questions asked.

**Comment by RogerHungYi** [comment]
As I tap purchase again, it shows something like "update is no further charge"! Therefore I reload app to my watch again without pay any extra money!
According to your earlier suggestion, I turn off power save mode but I still can not login "watch for tesla"!
But one thing is odd, as I said previously, I can not find "Apps only on watch" in iPhone setting!!! Is it might be the problem which cause "watch for tesla"won't work?

**Comment by Kemmey** [comment]
@RogerHungYi It might be related, yes. But only when your phone is on cellular. 

Have you tried while on WiFi?

Are you by any chance running a VPN? And if so, could you try without?

I'm really grabbing at straws here - I've also asked a few other Tesla devs for ideas - nothing comes to mind. It's so frustrating!

**Comment by RogerHungYi** [comment]
@Kemmey I have tried both on 4G and WiFi already! and no VPN running in my phone!
I know my problem cause you headache, sorry for that, but I still hope that when I switch from series3 to new applewatchS6, all the problems can be resolved!(cross finger)

Cheer up! It is not end of the world! I know you tried very hard to help me solving this strange problem and I'm really appreciate that! Thanks!!

**Comment by RogerHungYi** [comment]
![image](https://user-images.githubusercontent.com/70886708/94680245-04f17000-0354-11eb-985d-1e6eb416dbab.jpeg)

**Comment by RogerHungYi** [comment]
@Kemmey As I tap watch app for tesla again, it shows different message as I posted above! Can you tell any difference?!

Thanks

**Comment by Kemmey** [comment]
Thats just the release notes for new version. Dismiss it in top left corner

**Comment by RogerHungYi** [comment]
@Kemmey 
Good news, as I switch my apple watch from S3 to new S6, and connect to my tesla, it finally works!!!

Thank for your help, and I hope my experience can be helpful for others!

**Comment by Kemmey** [comment]
@RogerHungYi wow, that is fantastic news! But I do wonder what the problem was with the s3 watch - it made absolutely no sense, but awesome that you finally got it working! Thank you so much for holding out for such a long time!
